Delete all trivial uses of make_variable.

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29667 Delete redefinitions of methods in Variable already present on Tensor.
* #29665 Move most methods off Variable into torch::autograd::impl functions.
* #29653 Remove TensorImpl::is_variable, deprecate Tensor::is_variable
* **#29213 Delete all trivial uses of make_variable.**
* #29204 Miscellaneous follow up for code review comments
* #29203 Rename getNonVariableDeprecatedTypeProperties to getDeprecatedTypeProperties

A trivial use of make_variable is one where requires_grad=False.  This
transformation is not technically semantics preserving, as make_variable
will create a shallow copy of the tensor in question; however, I
am guessing that we have the invariant that we don't actually make
use of this shallow copy in a nontrivial way.

There were some cases where the surrounding code expected a Variable proper
to be returned; I retained those sites.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18353503](https://our.internmc.facebook.com/intern/diff/D18353503)

ezyang committed:
Delete all trivial uses of make_variable.

A trivial use of make_variable is one where requires_grad=False.  This
transformation is not technically semantics preserving, as make_variable
will create a shallow copy of the tensor in question; however, I
am guessing that we have the invariant that we don't actually make
use of this shallow copy in a nontrivial way.

There were some cases where the surrounding code expected a Variable proper
to be returned; I retained those sites.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

pietern commented:
@ezyang Can you confirm that the issue in #19145 is no longer valid given the recent merge?

ezyang commented:
@pietern confirmed

kostmo commented:
## CircleCI build failures summary
As of commit 2c27808:
* 0/1 recognized as flaky
* 1/1 failures introduced in this PR



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=2c278083253f679c2b7fae1bf05080c36b4a948d).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

ezyang commented:
Failures are probably real.

ezyang committed:
Update on "Delete all trivial uses of make_variable."


A trivial use of make_variable is one where requires_grad=False.  This
transformation is not technically semantics preserving, as make_variable
will create a shallow copy of the tensor in question; however, I
am guessing that we have the invariant that we don't actually make
use of this shallow copy in a nontrivial way.

There were some cases where the surrounding code expected a Variable proper
to be returned; I retained those sites.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Delete all trivial uses of make_variable."


A trivial use of make_variable is one where requires_grad=False.  This
transformation is not technically semantics preserving, as make_variable
will create a shallow copy of the tensor in question; however, I
am guessing that we have the invariant that we don't actually make
use of this shallow copy in a nontrivial way.

There were some cases where the surrounding code expected a Variable proper
to be returned; I retained those sites.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Delete all trivial uses of make_variable."


A trivial use of make_variable is one where requires_grad=False.  This
transformation is not technically semantics preserving, as make_variable
will create a shallow copy of the tensor in question; however, I
am guessing that we have the invariant that we don't actually make
use of this shallow copy in a nontrivial way.

There were some cases where the surrounding code expected a Variable proper
to be returned; I retained those sites.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18353503](https://our.internmc.facebook.com/intern/diff/D18353503)

[ghstack-poisoned]

ezyang committed:
Update on "Delete all trivial uses of make_variable."


A trivial use of make_variable is one where requires_grad=False.  This
transformation is not technically semantics preserving, as make_variable
will create a shallow copy of the tensor in question; however, I
am guessing that we have the invariant that we don't actually make
use of this shallow copy in a nontrivial way.

There were some cases where the surrounding code expected a Variable proper
to be returned; I retained those sites.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D18353503](https://our.internmc.facebook.com/intern/diff/D18353503)

[ghstack-poisoned]

yf225 commented on torch/csrc/cuda/comm.cpp:
```diff
@@ -86,10 +86,12 @@ std::vector<Tensor> broadcast(const Tensor& tensor, IntArrayRef devices) {
-// We thus re-wrap these Variables after broadcasting (i.e., effetively doing
+// We thus re-wrap these Variables after broadcasting (i.e., effectively doing
 // what is equivalent to .data in Python), and give them individual version
 // counters.
```
Based on this comment, would it work if we just call `var.data()` instead of keeping the original `make_variable(var.tensor_data(), false)`?

ezyang replied:
Maybe. I tried something similar and it didn't work; I'm going to have to study this site more carefully. Leaving for another patch.
